chia/wallet/coin_selection.py

```diff
@@ -0,0 +1,62 @@
+import random
+from typing import Set, Optional, Tuple, List
+
+from chia.consensus.default_constants import DEFAULT_CONSTANTS
+from chia.types.blockchain_format.coin import Coin
+from chia.util.ints import uint64
+
+# These algorithms were based off of the algorithms in:
+# https://murch.one/wp-content/uploads/2016/11/erhardt2016coinselection.pdf
+
+
+# we use this to check if one of the coins exactly matches the target.
+def check_for_exact_match(coin_list: List[Coin], target: uint64) -> Optional[Coin]:
+    for coin in coin_list:
+        if coin.amount == target:
+            return coin
+    return None
+
+
+# we use this to find an individual coin greater than the target but as close as possible to the target.
+def find_smallest_coin(greater_coin_list: List[Coin], target: uint64) -> Optional[Coin]:
+    smallest_value = DEFAULT_CONSTANTS.MAX_COIN_AMOUNT  # smallest coins value

```

Comment:
Usually using DEFAULT_CONSTANTS directly in the code is not great because if you want to run a testnet with modified constants, for example, you want the whole code to use your new constants. The better way is passing them in (the wallet.wallet_state_manager.constants.). However it's not a big deal here since this constant will never change in forks or testnets